### PR TITLE
[docs] Asset storage description in filesystem IO Manager docs

### PR DESCRIPTION
**Pull request by clairelin135**
Updates IO manager docs to describe how filesystem IO managers store assets.

### CODE REVIEW

File: docs/content/concepts/io-management/io-managers.mdx [hunk 1]
```diff
@@ -50,7 +50,9 @@ height={1040}
 
 This diagram shows a job with two IO managers, each of which is shared across a few inputs and outputs.
 
-The default IOManager, <PyObject module="dagster" object="fs_io_manager" />, stores and retrieves values in the filesystem while pickling. If a job is invoked via <PyObject object="JobDefinition" method="execute_in_process" />, the default IOManager is switched to <PyObject module="dagster" object="mem_io_manager"/>, which stores outputs in memory. Dagster provides out-of-the-box IOManagers that pickle objects and save them. These are <PyObject module="dagster_aws.s3" object="s3_pickle_io_manager"/> , <PyObject module="dagster_azure.adls2" object="adls2_pickle_io_manager"/> , or <PyObject module="dagster_gcp.gcs" object="gcs_pickle_io_manager"/>.
+The default IOManager, <PyObject module="dagster" object="fs_io_manager" />, stores and retrieves values in the filesystem while pickling. If a job is invoked via <PyObject object="JobDefinition" method="execute_in_process" />, the default IOManager is switched to <PyObject module="dagster" object="mem_io_manager"/>, which stores outputs in memory.
```
Comment (erinkcochran87) [inline docs/content/concepts/io-management/io-managers.mdx:53]: This page uses `IO manager` and `IO managers`, and your updates use those without a space (`IOManager`). I don't have a preference for which one we use as long as it's consistent - unless this is referring to something else entirely?

File: python_modules/libraries/dagster-aws/dagster_aws/s3/io_manager.py [hunk 2]
```diff
@@ -90,6 +90,16 @@ def s3_pickle_io_manager(init_context):
     Serializes objects via pickling. Suitable for objects storage for distributed executors, so long
     as each execution node has network connectivity and credentials for S3 and the backing bucket.
 
+    Assigns each op output to a unique filepath containing run ID, step key, and output name.
+    Assigns each asset to a single filesystem path, at "<base_dir>/<asset_key>". If the asset key
+    has multiple components, the final component is used as the name of the file, and the preceding
+    components as parent directories under the base_dir.
+
+    Subsequent materializations of an asset will overwrite previous materializations of that asset.
+    So, with a base directory of "/my/base/path", an asset with key
```
Comment (erinkcochran87) [inline python_modules/libraries/dagster-aws/dagster_aws/s3/io_manager.py:99]: Nit:
```suggestion
    With a base directory of "/my/base/path", an asset with key
```

File: python_modules/libraries/dagster-azure/dagster_azure/adls2/io_manager.py [hunk 3]
```diff
@@ -116,6 +116,16 @@ def adls2_pickle_io_manager(init_context):
     as each execution node has network connectivity and credentials for ADLS and the backing
     container.
 
+    Assigns each op output to a unique filepath containing run ID, step key, and output name.
+    Assigns each asset to a single filesystem path, at "<base_dir>/<asset_key>". If the asset key
+    has multiple components, the final component is used as the name of the file, and the preceding
+    components as parent directories under the base_dir.
+
+    Subsequent materializations of an asset will overwrite previous materializations of that asset.
+    So, with a base directory of "/my/base/path", an asset with key
```
Comment (erinkcochran87) [inline python_modules/libraries/dagster-azure/dagster_azure/adls2/io_manager.py:125]: Same here:
```suggestion
    With a base directory of "/my/base/path", an asset with key
```

File: python_modules/libraries/dagster-gcp/dagster_gcp/gcs/io_manager.py [hunk 4]
```diff
@@ -89,6 +89,16 @@ def gcs_pickle_io_manager(init_context):
     Serializes objects via pickling. Suitable for objects storage for distributed executors, so long
     as each execution node has network connectivity and credentials for GCS and the backing bucket.
 
+    Assigns each op output to a unique filepath containing run ID, step key, and output name.
+    Assigns each asset to a single filesystem path, at "<base_dir>/<asset_key>". If the asset key
+    has multiple components, the final component is used as the name of the file, and the preceding
+    components as parent directories under the base_dir.
+
+    Subsequent materializations of an asset will overwrite previous materializations of that asset.
+    So, with a base directory of "/my/base/path", an asset with key
```
Comment (erinkcochran87) [inline python_modules/libraries/dagster-gcp/dagster_gcp/gcs/io_manager.py:98]: lol, one more time:
```suggestion
    With a base directory of "/my/base/path", an asset with key
```

**Review by erinkcochran87 — APPROVED** [review]
This looks good! I suggested some *super* minor changes, so I'm stamping this as approved :shipit:

File: docs/content/concepts/io-management/io-managers.mdx (hunk 1 above)
Comment (clairelin135) [reply docs/content/concepts/io-management/io-managers.mdx:53]: ahh, nope this is referring to the same thing! I will change my update to say `IO manager`

**Pull request merged**